src/GraphQL/Execution/GraphQLDocumentBuilder.cs

```diff
@@ -20,7 +20,7 @@ public class GraphQLDocumentBuilder : IDocumentBuilder
         public bool IgnoreComments { get; set; } = true;
 
         /// <inheritdoc/>
-        public Document Build(string body)
+        public IParseResult Build(string body)
```

Comment:
Why change the signature of this method? There will always be no more than 1 error. Just add `try/catch` in the calling method.